components/ui/common/SwipeableRow/options/VoteOptions.tsx

```diff
@@ -0,0 +1,239 @@
+import Animated, {
+  Extrapolation,
+  interpolate,
+  interpolateColor,
+  runOnJS,
+  useAnimatedReaction,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withSpring,
+} from "react-native-reanimated";
+import { trigger } from "react-native-haptic-feedback";
+import { StyleSheet, ColorValue, LayoutRectangle, Alert } from "react-native";
+import { useEffect, useRef, useState } from "react";
+import { useSwipeableRow } from "../SwipeableRowProvider";
+import { IconArrowUp } from "tabler-icons-react-native";
+import { ILemmyVote } from "../../../../../lemmy/types/ILemmyVote";
+import { getUniqueID } from "@ronradtke/react-native-markdown-display";
+
+interface VoteColor {
+  background: ColorValue;
+}
+
+type VoteColors = {
+  upvote: VoteColor;
+  downvote: VoteColor;
+};
+
+const DEFAULT_VOTE_COLORS: VoteColors = {
+  upvote: {
+    background: "orange",
+  },
+  downvote: {
+    background: "blue",
+  },
+};
+
+type Stops = [first: number, second: number];
+const DEFAULT_STOPS: Stops = [75, 125];
+
+interface Props {
+  stops?: Stops;
+  voteColors?: VoteColors;
+  vote?: number;
+  onUpVote: () => void;
+  onDownVote: () => void;
+}
+
+const buzz = () => {
+  "worklet";
+  runOnJS(trigger)("medium");
+};
+
+export const VoteOption = ({
+  stops = DEFAULT_STOPS,
+  voteColors = DEFAULT_VOTE_COLORS,
+  vote = 0,
+  onUpVote,
+  onDownVote,
+}: Props) => {
+  const [firstStop, secondStop] = stops;
+  const [first, second] =
+    vote === -1
+      ? [voteColors.downvote, voteColors.upvote]
+      : [voteColors.upvote, voteColors.downvote];
+
+  // Provides UI thread access to the current vote as it changes.
+  const currentVote = useDerivedValue<number>(() => vote);
```

Comment:
In order for the vote to be always up to date on the main thread i use a reanimated derived value. I'm not convinced I need this now that `subscribers` is stateful, but i haven't validated that yet.